Modifying the script into a function

GeorgeYQian opened the pull request.

GeorgeYQian committed:
Modifying the script into a function

GeorgeYQian committed:
Add files via upload

thibautjombart commented on R/branching_basics.R:
```diff
@@ -1,175 +1,171 @@
+# Draw individual R based on group values and frequencies
+# n: number of R values to draw
+draw_R <- function(n,
+                   group_R = c(R_detected, R_undetected),
+                   group_p = c(p_detected, 1 - p_detected)
```
These will not be defined now as these variables only exist within `branching_process_model`; best remove the defaults from here, and pass the values when calling the function inside `branching_process_model`.

thibautjombart commented on R/branching_basics.R:
```diff
@@ -1,175 +1,171 @@
+
+## Determine R for each case
+out <- mutate(out,
+              R = draw_R(nrow(out))
```
As defaults of `draw_R` will be removed, pass the values here.

thibautjombart commented on R/branching_basics.R:
```diff
@@ -1,175 +1,171 @@
+} #end function here
+
+
+# ------------
+# Check output
+# ------------
+library(incidence2)
+out %>%
+  incidence(date_onset, interval = 7) %>%
+  plot(title = "Simulated incidence", xlab = "Time")
```
Replace this bit by an example, i.e. generate a simulation to replace `out`.

thibautjombart commented on branching_basics2.R:
```diff
@@ -0,0 +1,227 @@
+
```
Not sure what this second file is for; I suspect it may need to be removed, if this is only for backup purpose.

thibautjombart commented on run_branching_basics.R:
```diff
@@ -0,0 +1,82 @@
+library(epitrix)
+library(here)
+library(tidyverse) # total overkill, we merely use tibble and dplyr
+library(foreach)
+library(doParallel)
+registerDoParallel(cores=3)
+
+
+## WORKAROUND: https://github.com/rstudio/rstudio/issues/6692
+## Revert to 'sequential' setup of PSOCK cluster in RStudio Console on macOS and R 4.0.0
+if (Sys.getenv("RSTUDIO") == "1" && !nzchar(Sys.getenv("RSTUDIO_TERM")) && 
+    Sys.info()["sysname"] == "Darwin" && getRversion() >= "4.0.0") {
+  parallel:::setDefaultClusterOptions(setup_strategy = "sequential")
+}
+
```
I think we discussed using `future.lapply` - is this still used?

thibautjombart reviewed (changes requested):
Thanks, this is moving into the right direction. Key issues to address are:

* remove files which are not needed if they have been committed to the history tree before (we don't need backups, git handles that for us)
* fix the defaults in `draw_R` (see comments)
* check that the script running simulations is up to date, or even remove it; it is easier to handle a small PR doing one thing rather than addressing several points at the same time.

GeorgeYQian committed:
Update branching_basics2.R

GeorgeYQian committed:
Update run_branching_basics.R

GeorgeYQian committed:
Update run_branching_basics.R

Add max and minimum number of cases expected during each simulation (for the ABC)

GeorgeYQian committed:
Create ABC_basic.R

File for performing ABC on our results

GeorgeYQian committed:
Update ABC_basic.R

GeorgeYQian committed:
Create Marburg_Results

Store Marburg branching process model results

GeorgeYQian committed:
Delete Marburg_Results

GeorgeYQian committed:
Create First_File

First file (dummy)

GeorgeYQian committed:
FirstDummy

GeorgeYQian committed:
Add files via upload

GeorgeYQian committed:
Create Dummy

GeorgeYQian committed:
Add files via upload

GeorgeYQian committed:
Create Dummy

GeorgeYQian committed:
Add files via upload

GeorgeYQian committed:
Create dummy

GeorgeYQian committed:
Add files via upload

GeorgeYQian committed:
Update ABC_basic.R

GeorgeYQian committed:
Update branching_basics2.R

GeorgeYQian committed:
Create R_Int_Plotly

GeorgeYQian committed:
Update run_branching_basics.R

GeorgeYQian committed:
Create R_Int_Plotly_Figure.JPG

GeorgeYQian committed:
Merge branch 'branch_george' of https://github.com/thibautjombart/branching_basics into branch_george

GeorgeYQian committed:
Update: Modified branching_basics2 to get rid of stuff that was slowing down the code (legacy + foreach stuff)  Modified run_branching_basicsV2 so that it can run the model and perform a basic ABC given these changes

GeorgeYQian committed:
This file incorporates ring vaccination

This file incorporates ring vaccination

GeorgeYQian committed:
Update branching_basics_ring_vacc.R

GeorgeYQian committed:
Update 20th December

thibautjombart committed:
fixing issue in non-defined SI

thibautjombart committed:
more fixes but still not working

GeorgeYQian committed:
Update to tidy up code and ensure it works

GeorgeYQian committed:
Update: have cases been reported?

thibautjombart committed:
various fixes; vaccination + reporting now works

GeorgeYQian committed:
Update: Include different vaccine efficacy, tidy up logistic function

GeorgeYQian committed:
Create Logistic functions for vaccine efficacy approximation

GeorgeYQian committed:
Upload updated model files
